apis/python/src/tiledbsoma/utils/IntIndexer.py

```diff
@@ -0,0 +1,21 @@
+from typing import Union
+
+import numpy as np
+
+from tiledbsoma import pytiledbsoma as clib
+from tiledbsoma._experiment import _indexer_map_locations
+from tiledbsoma.options import SOMATileDBContext
+
+"""External API"""
+
+
+class IntIndexer:
+    @staticmethod
+    def map_locations(
+        keys: np.typing.NDArray[np.int64],
+        context: Union[SOMATileDBContext, None] = None,
+    ) -> clib.IntIndexer:  # type ignore
+        return _indexer_map_locations(keys, context)
```

Comment:
What I was trying to suggest in the last round of review is that there is no need for a `IntIndexer.map_locations` function; the `indexer_map_locations` function should be exported directly without wrapping it in a class.

Assuming we want to export this name on the `utils` namespace (which makes sense to me), I think the best way to accomplish this would be something like:

```python
# in tiledbsoma/utils.py

# This is a naming suggestion; 
def build_index(keys: ..., context: ...) -> clib.IntIndexer:
  # the contents of _indexer_map_locations above
```

```python
# in tiledbsoma/_experiment.py
from . import utils

class Experiment(...):
  def axis_query(...):
    return query.ExperimentAxisQuery(
        ...,
        index_factory=functools.partial(utils.build_index, ...),
    )
```

This will make the function available to end users, with its home in the `tiledbsoma.utils` namespace.

Reply:
Yeah, from my perspective, a static class function and a global function sounds the same (me being naive). I like this model and if @bkmartinjr has no objection I'll go ahead with your suggestion @thetorpedodog.

Reply:
Make the suggested changes. Plz lmk if it all looks good @thetorpedodog